src/components/cluster-deployments/ClusterDeploymentDetailsPage.tsx

```diff
@@ -0,0 +1,172 @@
+  const [agentClusterInstall, agentClusterInstallLoaded, agentClusterInstallError] =
+    useK8sWatchResource<CIM.AgentClusterInstallK8sResource>(
+      clusterDeployment?.spec?.clusterInstallRef.name
+        ? {
+            kind: AgentClusterInstallKind,
+            name: clusterDeployment?.spec?.clusterInstallRef.name,
+            namespace: clusterDeployment?.metadata?.namespace,
+            namespaced: true,
+            isList: false,
+          }
+        : undefined,
+    );
+
+  const agentSelector = clusterDeployment.spec?.platform?.agentBareMetal?.agentSelector;
+  const [agents, agentsLoaded, agentsError] = useK8sWatchResource<CIM.AgentK8sResource[]>(
+    agentSelector
+      ? {
+          kind: AgentKind,
+          isList: true,
+          selector: agentSelector,
+          namespaced: true,
+        }
+      : undefined,
+  );
+
+  if (!clusterDeployment) return null;
+  if (agentsError) throw new Error(agentsError);
+  if (agentClusterInstallError) throw new Error(agentClusterInstallError);
+  if (!(agentsLoaded && agentClusterInstallLoaded)) return <LoadingState />;
+
+  const handleKubeconfigDownload = async () => {
+    const kubeconfigSecretName =
+      agentClusterInstall.spec?.clusterMetadata?.adminKubeconfigSecretRef.name;
+    const kubeconfigSecretNamespace = clusterDeployment.metadata.namespace;
+    agentClusterInstall.spec?.clusterMetadata?.adminKubeconfigSecretRef.name;
+    try {
+      const kubeconfigSecret = await k8sGet(
+        SecretModel,
+        kubeconfigSecretName,
+        kubeconfigSecretNamespace,
+      );
+      const blob = new Blob([atob(kubeconfigSecret.data.kubeconfig)], {
+        type: 'text/plain;charset=utf-8',
+      });
+      saveAs(blob, 'kubeconfig.json');
+    } catch (e) {
+      console.error('Failed to fetch kubeconfig secret.', e);
+    }
+  };
+
+  const handleFetchEvents = async (props, onSuccess, onError) => {
+    try {
+      const res = await fetch(agentClusterInstall.status?.debugInfo?.eventsURL, {
+        mode: 'same-origin',
+      });
+      const data = await res.json();
+      onSuccess(
+        data.map((event: any) => ({
+          clusterId: event.cluster_id,
+          eventTime: event.event_time,
+          message: event.message,
+          severity: event.severity,
+        })),
+      );
+    } catch (e) {
+      onError('Failed to fetch cluster events.');
+    }
+  };
+
+  const handleFetchCredentials = async (setCredentials) => {
+    const adminPasswordSecretRefName =
+      agentClusterInstall.spec?.clusterMetadata?.adminPasswordSecretRef.name;
+    const namespace = clusterDeployment.metadata.namespace;
+
+    try {
+      const secret = await k8sGet(SecretModel, adminPasswordSecretRefName, namespace);
+      setCredentials({
+        username: atob(secret.data.username),
+        password: atob(secret.data.password),
+      });
+    } catch (e) {
+      console.error('Failed to fetch adminPasswordSecret secret.', e);
+    }
+  };
```

Comment:
when we get `fetchSecret` as mentioned above, this whole thing can be done in ai lib

Reply:
Done